src/many-ledger/src/module.rs

```diff
@@ -297,3 +377,211 @@ impl ManyAbciModuleBackend for LedgerModuleImpl {
+        })
+    }
+
+    fn get_roles(
+        &self,
+        _sender: &Identity,
+        args: GetRolesArgs,
+    ) -> Result<GetRolesReturn, ManyError> {
+        let account = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        let mut roles = BTreeMap::new();
+        for id in args.identities {
+            roles.insert(id, account.get_roles(&id));
+        }
+
+        Ok(GetRolesReturn { roles })
+    }
+
+    fn add_roles(
+        &mut self,
+        sender: &Identity,
+        args: AddRolesArgs,
+    ) -> Result<EmptyReturn, ManyError> {
+        let mut account = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        if !account.has_role(sender, "owner") {
+            return Err(account::errors::user_needs_role("owner"));
+        }
+        for (id, roles) in args.roles {
+            for r in roles {
+                account.add_role(&id, r);
+            }
+        }
+
+        self.storage.commit_account(&args.account, account)?;
+        Ok(EmptyReturn)
+    }
+
+    fn remove_roles(
+        &mut self,
+        sender: &Identity,
+        args: RemoveRolesArgs,
+    ) -> Result<EmptyReturn, ManyError> {
+        let mut account = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        if !account.has_role(sender, "owner") {
+            return Err(account::errors::user_needs_role("owner"));
+        }
+        for (id, roles) in args.roles {
+            for r in roles {
+                account.remove_role(&id, r);
+            }
+        }
+
+        self.storage.commit_account(&args.account, account)?;
+        Ok(EmptyReturn)
+    }
+
+    fn info(&self, _sender: &Identity, args: InfoArgs) -> Result<InfoReturn, ManyError> {
+        let Account {
+            description,
+            roles,
+            features,
+        } = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        Ok(InfoReturn {
+            description,
+            roles,
+            features,
+        })
+    }
+
+    fn delete(&mut self, sender: &Identity, args: DeleteArgs) -> Result<EmptyReturn, ManyError> {
+        let account = self
+            .storage
+            .get_account(&args.account)
+            .ok_or_else(|| account::errors::unknown_account(args.account))?;
+
+        if !account.has_role(sender, "owner") {
+            return Err(account::errors::user_needs_role("owner"));
+        }
+
+        self.storage.delete_account(&args.account)?;
+        Ok(EmptyReturn)
+    }
+
+    fn add_features(
+        &mut self,
+        _sender: &Identity,
+        _args: AddFeaturesArgs,
+    ) -> Result<EmptyReturn, ManyError> {
```

Comment:
I'll only comment it once, but I would like to see `pub type AddFeaturesReturn = EmptyReturn` and use that

Reply:
will do.